### PR TITLE
Suppress verbose MIR comments for trivial types

### PR DESCRIPTION
**Pull request by alasher**
Addresses #74508

This is my first contribution to the Rust project! Please let me know if anything needs revising, I'm happy to make changes.

### CODE REVIEW

**Comment by lcnr** [comment]
:+1: r? @oli-obk 

You still have to run `./x.py test src/test/mir-opt --stage 1 --bless` to update the test outputs which use pretty printing.

The same also goes for 32 bit mir tests, let me quickly find the command for that again :sweat_smile:

**Comment by lcnr** [comment]
For the remaining test I think you can use `./x.py test src/test/mir-opt/ --bless --stage 1 --target i686-unknown-linux-gnu`. You might have to install the multilib version of your toolchain to do so.

**Comment by alasher** [comment]
Thanks for the command to run the tests, I figured I'd run into trouble with those but didn't know how to run them. 😄 I've rebased the modified test files into the same commit, let me know if they should be split. I'll reupdate the test files if necessary after review.

**Comment by lcnr** [comment]
I think keeping them split would be useful, as the test changes are quite large :thinking:

Didn't look too much at the tests yet, but that's already looking a lot better IMO

File: src/librustc_mir/util/pretty.rs [hunk 1]
```diff
@@ -387,20 +387,30 @@ impl Visitor<'tcx> for ExtraComments<'tcx> {
     fn visit_constant(&mut self, constant: &Constant<'tcx>, location: Location) {
         self.super_constant(constant, location);
         let Constant { span, user_ty, literal } = constant;
-        self.push("mir::Constant");
-        self.push(&format!("+ span: {}", self.tcx.sess.source_map().span_to_string(*span)));
-        if let Some(user_ty) = user_ty {
-            self.push(&format!("+ user_ty: {:?}", user_ty));
+        match literal.ty.kind {
+            ty::Int(_) | ty::Uint(_) | ty::Bool | ty::Char => (),
```
Comment (lzutao) [inline src/librustc_mir/util/pretty.rs:391]:
```suggestion
            ty::Int(_) | ty::Uint(_) | ty::Bool | ty::Char => {},
```

File: src/librustc_mir/util/pretty.rs [hunk 2]
```diff
@@ -387,20 +387,30 @@ impl Visitor<'tcx> for ExtraComments<'tcx> {
...
     }
 
     fn visit_const(&mut self, constant: &&'tcx ty::Const<'tcx>, _: Location) {
         self.super_const(constant);
         let ty::Const { ty, val, .. } = constant;
-        self.push("ty::Const");
-        self.push(&format!("+ ty: {:?}", ty));
-        self.push(&format!("+ val: {:?}", val));
+        match ty.kind {
+            ty::Int(_) | ty::Uint(_) | ty::Bool | ty::Char => (),
```
Comment (lzutao) [inline src/librustc_mir/util/pretty.rs:407]:
```suggestion
            ty::Int(_) | ty::Uint(_) | ty::Bool | ty::Char => {},
```

**Comment by alasher** [comment]
@lzutao Addressed your feedback + rebased

**Comment by oli-obk** [comment]
@bors r+ p=6 bitrotty

Thanks! so much better

**Comment by oli-obk** [comment]
Oh no... `instrument_coverage` failed. In order to run that test you need to enable the profiler flag: https://github.com/rust-lang/rust/blob/e8df0b893235aa06c6362d1a8646296310f79f81/config.toml.example#L220

**Comment by alasher** [comment]
@oli-obk Alright, I think we're good now! `instrument_coverage` has been updated, unless there's another testcase hiding somewhere we should be good :) I `rg`'d the whole `mir-opt` folder and couldn't find anything else matching the now-removed output.

**Pull request merged**